diff --git a/src/lte/model/lte-mi-error-model.cc b/src/lte/model/lte-mi-error-model.cc
index fd9a2f1..b8ac38e 100644
--- a/src/lte/model/lte-mi-error-model.cc
+++ b/src/lte/model/lte-mi-error-model.cc
@@ -31,7 +31,7 @@
 
 
 #include <list>
-#include <tr1/functional>
+#include <functional>
 #include <vector>
 #include <ns3/log.h>
 #include <ns3/pointer.h>
diff --git a/src/lte/model/rr-ff-mac-scheduler.cc b/src/lte/model/rr-ff-mac-scheduler.cc
index 1ca9955..f6683f8 100644
--- a/src/lte/model/rr-ff-mac-scheduler.cc
+++ b/src/lte/model/rr-ff-mac-scheduler.cc
@@ -978,7 +978,7 @@ RrFfMacScheduler::DoSchedDlTriggerReq (const struct FfMacSchedSapProvider::Sched
   // Divide the resource equally among the active users according to
   // Resource allocation type 0 (see sec 7.1.6.1 of 36.213)
 
-  int rbgPerTb = (nTbs > 0) ? ((rbgNum - rbgAllocatedNum) / nTbs) : DBL_MAX;
+  int rbgPerTb = (nTbs > 0) ? ((rbgNum - rbgAllocatedNum) / nTbs) : static_cast<int>(DBL_MAX);
   NS_LOG_INFO (this << " Flows to be transmitted " << nflows << " rbgPerTb " << rbgPerTb);
   if (rbgPerTb == 0)
     {
diff --git a/src/netanim/model/animation-interface.h b/src/netanim/model/animation-interface.h
index 23e89c2..49ba588 100644
--- a/src/netanim/model/animation-interface.h
+++ b/src/netanim/model/animation-interface.h
@@ -58,7 +58,7 @@ typedef struct
 
 struct LinkPairCompare
 {
-  bool operator () (P2pLinkNodeIdPair first, P2pLinkNodeIdPair second)
+  bool operator () (P2pLinkNodeIdPair first, P2pLinkNodeIdPair second) const
     {
       //Check if they are the same node pairs but flipped
       if (  ((first.fromNode == second.fromNode) && (first.toNode == second.toNode)) ||
